Paper 58

atrautsch opened the pull request.

Commit:
init

Commit:
remove comment

Commit:
up

Commit:
up

Commit:
update descriptions

Commit:
add clarification that not everything is needed for replicating the paper

Commit:
bolded

Commit:
fix link

sherbold committed:
Update README.md

sherbold committed:
Update INSTALL.md

sherbold committed:
Update STATUS.md

atrautsch committed:
Merge pull request #1 from sherbold/master

Smaller changes

Commit:
clarification

Commit:
add second folder

anonymousLw commented:
Dear authors,

Thank you very much for your efforts in preparing the artifact.

I successfully run the scripts in Jupyter following the instructions in INSTALL. However, I found there's a lack of documentation for the file structure of the folders in the replication package. I am not sure where to find the generated results. Would you please add some explanation to the folders in the replication package? Also please provide some explanations for the different csv files as well as the heads of each column in the csv files if possible.

I would also appreciate if you can highlight those folders/files/columns that people need to pay attention to when replicating your results.

Thanks!

Commit:
add folder and CSV field information

Commit:
clarify results location

atrautsch commented:
Dear Reviewer,

thank you for your time!
I added folder structure documentation as well as CSV header info in the README.md. As the raw mining data CSV contains a lot it is a little cluttered I hope it is understandable. 
I also added a little clarification for the location of the result plots in the INSTALL.md.

I hope this clarifies everything. Having worked a lot with the data one gets a little blind to the usability side of things.

klainfo commented:
Thank you for submitting the artefact to the ICSME'20 Artifact/Rose Track. Congratulations! Three reviewers agreed that the artefact is reusable and available. 

-----------------------------
Submission ID: # 58
Decision: Accept
Recommended for Best Artifact Awards: No
Score (out of 10): 8

Comments:
The artifact is well-prepared and easy to use. The authors also provided a DOI for the artifact. Following the steps specified in INSTALL.md, I can easily install the dependencies and run the code with Jupyter notebook. With the code I can reproduce the figures in Figure 2-6 in the paper and data in Table 6 and 8. These are the major results of the paper. I recommend assigning the paper both available and reusable badges.

By comparing the results generated by the code and the results reported in the paper, I found that there are some potential typos in Table 8.
- train/test split, ad-hoc SZZ label, combined feature set, 1/2(#LR+#RF) should be 28.5=(26+31)/2
- The last row 99 and 143 should be swapped. #LR is 143 and #RF is 99

-----------------------------
Submission ID: # 58
Decision: Accept
Recommended for Best Artifact Awards: Ye
Score (out of 10): 8
Comments: 

Available and Reusable : the artifact is available on GitHub and fully functional. It is well documents and did not require so much effort to run. The authors used jupyter notebook, which was an elegant way for the particular artifact that they have developed, since this was data-driven. I was able to reproduce all figures and data files. 

The only weakness there was a python package that has been used to create a virtual environment which was not specified in the installation guide.

Anyway, I like this artifact, and I’d nominate it for the best artifact award

-----------------------------
Submission ID: #58
Decision: Accept
Recommended for Best Artifact Awards: No
Score (out of 10): 7
Comments:

I thank the authors for the thorough artifact submission. I was able to follow the instructions provided by the authors through the jupyter notebooks and also reproduce the figures of  the ICSME paper submission. I am recommending the following badges:

- The artifact is *available* as it is fully functional and will be placed on a public repository as promised by the authors.
- The artifact is *reusable* as it is functional and carefully documented. I also see that repurposing of the artifact is at reach. Given that all datasets and jupyter notebooks are available, repurposing these artifacts to one's research interests is easily feasible.
